devito/ir/equations/equation.py

```diff
@@ -154,12 +167,18 @@ def __new__(cls, *args, **kwargs):
         # Finally create the LoweredEq with all metadata attached
         expr = super(LoweredEq, cls).__new__(cls, expr.lhs, expr.rhs, evaluate=False)
 
+        expr._data = []
```

Comment:
this smells like a hack

Reply:
'Break' is a type of equation that will have specific properties that must be passed on to the LoweredEq to finally reach the visitors. These properties (namely 'condition' and 'else_expr') are specific to the Break class and I did not find any other way to pass such information further without creating an extra field to store equation specific data. The same is valid for the Reduce class.

Reply:
Yeah, I don't like this. @ArthurGiannotta lets talk about this at some stage to see what other implementations are possible.

Reply:
I will resolve this since the implementation completely changed.